src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private String lineSeparator = System.lineSeparator();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        validateInput(names, data, dateFrom, dateTo);
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int sum = 0;
+            for (int i = 0; i < data.length; i++) {
+                String[] dataLine = data[i].split(" ");
+                if (dataLine[1].equals(name)
+                        && LocalDate.parse(dataLine[0], FORMATTER)
+                            .compareTo(LocalDate.parse(dateFrom, FORMATTER)) >= 0
+                        && LocalDate.parse(dataLine[0], FORMATTER)
+                            .compareTo(LocalDate.parse(dateTo, FORMATTER)) <= 0) {
+                    sum += Integer.valueOf(dataLine[2]) * Integer.valueOf(dataLine[3]);
+                }
+            }
+            result.append(lineSeparator)
+                    .append(name)
+                    .append(" - ")
+                    .append(sum);
+        }
+        return result.toString();
+    }
```

Comment:
1. Hardcoded indexes are hard to read. Need constantly check source to remind what is that index
2. Too complicated conditions, need to extract to separate methods
3. May use static imports to avoid indication of class and thus reduce line length without readability impact
```suggestion
    for (String name : names) {
            int sum = 0;
            for (int i = 0; i < data.length; i++) {
                String[] dataLine = data[i].split(" ");
                if (dataLine[NAME_INDEX].equals(name)
                        && isSuitableDate(dataLine[DATE_INDEX], dateFrom, dateTo)) {
                    sum += parseInt(dataLine[HOURS_INDEX]) * parseInt(dataLine[HOURLY_WAGE_INDEX]);
                }
            }
            result.append(lineSeparator())
                    .append(name)
                    .append(" - ")
                    .append(sum);
        }
        return result.toString();
    }
    
    private boolean isSuitableDate(String date, String validFrom, String validTo) {
        return !parseDate(date).isBefore(parseDate(validFrom))
                && !parseDate(date).isAfter(parseDate(validTo))
    }
    
    private LocalDate parseDate(String dateAsString) {
        return LocalDate.parse(dateAsString, FORMATTER);
    }
```
+
import static java.lang.Integer.parseInt;
import static java.lang.System.lineSeparator;

+
constants